takari-lifecycle-plugin/src/test/java/io/takari/maven/plugins/compile/SimpleReactorReader.java

```diff
@@ -0,0 +1,156 @@
+package io.takari.maven.plugins.compile;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.artifact.ArtifactUtils;
+import org.apache.maven.model.Model;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.repository.internal.MavenWorkspaceReader;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.repository.WorkspaceRepository;
+import org.eclipse.aether.util.repository.ChainedWorkspaceReader;
+
+import com.google.common.collect.ImmutableMap;
+
+class SimpleReactorReader implements MavenWorkspaceReader {
+
```

Comment:
Yes, agree this should go to plugin testing harness. Eventually. Maybe. But right now this is the only place where I used this testing approach and I can't tell if it's generally useful or not yet.